lua/spectre/ui.lua

```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
         opts.live_update and '(Auto update)' or '',
         state.user_config.default.replace.cmd
     )
-    utils.write_virtual_text(state.bufnr, config.namespace_header, 0, { { help_text, state.user_config.highlight.headers } })
+    utils.write_virtual_text(state.bufnr, config.namespace_header, 0,
+        { { help_text, state.user_config.highlight.headers } })
 end
 
 M.show_menu_options = function(title, content)
     local win_width, win_height = vim.lsp.util._make_floating_popup_size(content, {})
-    local help_win, help_win_obj = popup.create(content, {
-        title   = title,
-        padding = { 1, 1, 1, 1 },
-        enter   = false,
-        width   = win_width + 2,
-        height  = win_height + 2,
-        col     = "cursor+2",
-        line    = "cursor+2",
-        border = true,
-        borderchars ={ "─", "│", "─", "│", "╭", "╮", "╯", "╰" },
+
+    local check = false
+    local bufnr = vim.api.nvim_create_buf(false, true)
+    api.nvim_buf_set_option(bufnr, "bufhidden", "wipe")
+    api.nvim_buf_set_lines(bufnr, 0, -1, true, content)
+
+    local help_win = vim.api.nvim_open_win(bufnr, true, {
+        title = title,
+        title_pos = 'center',
+        relative = 'cursor',
+        width = win_width + 3,
+        height = win_height + 2,
+        col = 4,
+        row = 4,
+        border = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" }
     })
+    api.nvim_win_set_option(help_win, 'winblend', 0)
+    api.nvim_win_set_option(help_win, 'number', false)
+    api.nvim_buf_set_keymap(bufnr, 'n', '<Esc>','<CMD>lua vim.api.nvim_win_close(' .. help_win  .. ', true)<CR>', {noremap = true})
 
-    vim.api.nvim_win_set_option(help_win, 'winblend', 0)
     api.nvim_create_autocmd({
-        'CursorMoved',
         'CursorMovedI',
```

Comment:
These should be added back to close the window if we disable automatically entering the help window (`vim.api.nvim_open_win(bufnr, false...`)
```suggestion
        'CursorMoved',
        'CursorMovedI',
        'BufHidden',
        'BufLeave',
        'InsertEnter',
        'WinScrolled',
        'BufDelete',
```

Reply:
It won't if it focus on window. I will add it back so it won't focus on it.